plugins/Diagnostics/Commands/ArchivingInstanceStatistics.php

```diff
@@ -0,0 +1,74 @@
+<?php
+/**
+ * Matomo - free/libre analytics platform
+ *
+ * @link https://matomo.org
+ * @license http://www.gnu.org/licenses/gpl-3.0.html GPL v3 or later
+ */
+
+namespace Piwik\Plugins\Diagnostics\Commands;
+
+use Piwik\Common;
+use Piwik\Date;
+use Piwik\Plugin\ConsoleCommand;
+use Piwik\Db;
+
+/**
+ * Diagnostic command that returns instance statistics related to archiving
+ */
+class ArchivingInstanceStatistics extends ConsoleCommand
+{
+    protected function configure()
+    {
+        $this->setName('diagnostics:archiving-instance-statistics');
+        $this->addNoValueOption('json', null,
+            "If supplied, the command will return data in json format");
+        $this->setDescription('Show data statistics which can affect archiving performance');
+    }
+
+    protected function doExecute(): int
+    {
+        $input = $this->getInput();
+        $output = $this->getOutput();
+
+        $metrics = $this->getArchivingInstanceStatistics();
+
+        if ($input->getOption('json')) {
+            $output->write(json_encode($metrics));
+        } else {
+            $headers = ['Statistic Name', 'Value'];
+            $this->renderTable($headers, $metrics);
+        }
+
+        return self::SUCCESS;
+    }
+
+    /**
+     * Retrieve various data statistics useful for diagnosing archiving performance
+     *
+     * @return array
+     */
+    public function getArchivingInstanceStatistics(): array
+    {
+        $stats = [];
+        $stats[] = ['Site Count', Db::fetchOne("SELECT COUNT(*) FROM " . Common::prefixTable("site"))];
```

Comment:
When _just_ counting rows, we were taught at uni it's best to count(1) and not all the columns, not sure it makes any difference here but I never had any issues with it so it's my go to.

```suggestion
        $stats[] = ['Site Count', Db::fetchOne("SELECT COUNT(1) FROM " . Common::prefixTable("site"))];
```

Can be applied in multiple places below.

Reply:
There's no difference between `COUNT(*)` and `COUNT(1)` when using InnoDB, there was an [optimization](https://dev.mysql.com/doc/refman/8.0/en/aggregate-functions.html#function_count) for `COUNT(1)` if using MyISAM but that doesn't apply here.

Reply:
Ok, all good. I wonder if it might still make sense to use `COUNT(1)`, especially if there's no difference and it can help on older engines. But we could look at that separately as it would probably be in multiple other places.

Reply:
We don't support MyISAM tables which are practically obsolete at this point.  Since the two statements are equivalent, there is no performance benefit on modern engines and `COUNT(*)` is a more common convention then it doesn't make sense to me to spend any effort changing this.

Reply:
Just a side note: A while back I read an article about that. For MySQL it indeed shouldn't make any difference, but for other databases it might. If I remember correctly for Postgres I think a `COUNT(1)` was 10% slower than a `COUNT(*)`, but might be possible that was improved there meanwhile.

Reply:
Could've been [this one](https://newrelic.com/blog/how-to-relic/fast-counting-in-postgresql-and-mysql), @sgiehl. Ok, TIL and will not raise this again ;-)